File: backend/src/Shalom.Application/Contracts/FastingOverrideDto.cs
namespace Shalom.Application.Contracts;

/// <summary>Per-weekday eating-window override (ADR-005: Sunday lunch is a first-class exception).</summary>
public record FastingOverrideDto(
    DayOfWeek DayOfWeek,
    TimeOnly EatingWindowStart,
    TimeOnly EatingWindowEnd);

File: frontend/projects/api/src/lib/models/fasting-override-dto.ts
/** Weekday name as serialized by the API ("Sunday" … "Saturday"). */
export type WeekdayName =
  | 'Sunday'
  | 'Monday'
  | 'Tuesday'
  | 'Wednesday'
  | 'Thursday'
  | 'Friday'
  | 'Saturday';

/**
 * Per-weekday eating-window override (ADR-005: Sunday family lunch is a
 * first-class exception, not a streak-breaker). Times are `HH:mm:ss`.
 */
export interface FastingOverrideDto {
  readonly dayOfWeek: WeekdayName;
  readonly eatingWindowStart: string;
  readonly eatingWindowEnd: string;
}
